include/boost/math/interpolators/detail/2d_akima_detail.hpp

```diff
@@ -0,0 +1,159 @@
+// Copyright Matt Borland, 2021
+// Use, modification and distribution are subject to the
+// Boost Software License, Version 1.0.
+// (See accompanying file LICENSE_1_0.txt
+// or copy at http://www.boost.org/LICENSE_1_0.txt)
+//
+// Source: https://www.its.bldrdoc.gov/DownloadPublications.ashx?id=OT-75-70.pdf
+
+#ifndef BOOST_MATH_INTERPOLATORS_2D_AKIMA_DETIAL_HPP
+#define BOOST_MATH_INTERPOLATORS_2D_AKIMA_DETIAL_HPP
+
+#include <type_traits>
+#include <stdexcept>
+#include <utility>
+#include <algorithm>
+#include <iostream>
+#include <cstddef>
+#include <cmath>
+
+namespace boost { namespace math { namespace interpolators { namespace detail {
+
+template <typename Matrix, typename VectorType = typename Matrix::value_type, typename Real = typename VectorType::value_type>
```

Comment:
The `Matrix::value_type` for Eigen matrices will be (say) `float` or `double`, not a vector.